tests/pipeline/test_engine.py

```diff
@@ -1242,6 +1251,241 @@ def test_dollar_volume(self):
+                    asset_column = int(asset) - 1
+                    expected_regression_results = linregress(
+                        y=other_asset_returns, x=my_asset_returns,
+                    )
+                    for i, output in enumerate(outputs):
+                        expected_output_results[output][day, asset_column] = \
+                            expected_regression_results[i]
+
+            for output in outputs:
+                assert_frame_equal(
+                    output_results[output],
+                    DataFrame(
+                        expected_output_results[output],
+                        index=self.dates[start_date_index:end_date_index + 1],
+                        columns=assets,
+                    ),
+                )
+
+    def test_correlation_and_regression_with_bad_asset(self):
+        """
+        Test that `RollingPearsonOfReturns`, `RollingSpearmanOfReturns` and
+        `RollingLinearRegressionOfReturns` raise the proper exception when
+        given a nonexistent target asset.
+        """
+        start_date_index = 6
+        end_date_index = 10
+        my_asset = Equity(0)
+
+        # This filter is arbitrary; the important thing is that we test each
+        # factor both with and without a specified mask.
+        my_asset_filter = AssetID().eq(1)
+
+        for mask in (NotSpecified, my_asset_filter):
+            pearson_factor = RollingPearsonOfReturns(
+                target=my_asset,
+                returns_length=3,
+                correlation_length=3,
+                mask=mask,
+            )
+            spearman_factor = RollingSpearmanOfReturns(
+                target=my_asset,
+                returns_length=3,
+                correlation_length=3,
+                mask=mask,
+            )
+            regression_factor = RollingLinearRegressionOfReturns(
+                target=my_asset,
+                returns_length=3,
+                regression_length=3,
+                mask=mask,
+            )
+
+            with self.assertRaises(NonExistentAssetInTimeFrame):
+                self.engine.run_pipeline(
+                    Pipeline(columns={'pearson_factor': pearson_factor}),
+                    self.dates[start_date_index],
+                    self.dates[end_date_index],
+                )
+            with self.assertRaises(NonExistentAssetInTimeFrame):
+                self.engine.run_pipeline(
+                    Pipeline(columns={'spearman_factor': spearman_factor}),
+                    self.dates[start_date_index],
+                    self.dates[end_date_index],
+                )
+            with self.assertRaises(NonExistentAssetInTimeFrame):
+                self.engine.run_pipeline(
+                    Pipeline(columns={'regression_factor': regression_factor}),
+                    self.dates[start_date_index],
+                    self.dates[end_date_index],
+                )
```

Comment:
Are these tests sufficient? A few questions:
- is there a better/more specific function call than `run_pipeline` on which we can assert the exception happens?
- should these test the actual exception string output?
- should this test the case of a partially existent asset? (not sure how to do this without modifying the asset finder fixture though)